ampersand-collection-rest-mixin.js

```diff
@@ -18,15 +9,23 @@ module.exports = {
     fetch: function(options) {
         options = options ? assign({}, options) : {};
         if (options.parse === void 0) options.parse = true;
+        var self = this;
+        
         var success = options.success;
-        var collection = this;
         options.success = function(resp) {
             var method = options.reset ? 'reset' : 'set';
-            if (options.set !== false) collection[method](resp, options);
-            if (success) success(collection, resp, options);
-            if (options.set !== false) collection.trigger('sync', collection, resp, options);
+            if (options.set !== false) self[method](resp, options);
+            if (success) success(self, resp, options);
+            if (options.set !== false) self.trigger('sync', self, resp, options);
+        };
+        
+        // Wrap an optional error callback with a fallback error event.
+        var error = options.error;
+        options.error = function(resp) {
+            if (error) error(self, resp, options);
+            self.trigger('error', self, resp, options);
         };
-        wrapError(this, options);
```

Comment:
In the past this wrapError has proven to be one abstraction over the line of what I was able to follow, and wholeheartedly agree w/ the direction taken here of just inlining it where needed.